script/core/FileManager.js

```diff
@@ -0,0 +1,44 @@
+export default class FileManager {
+	static loadFile(url, type) {
+		if (type === 'json') {
+			return loadJSON(url);
+		}
+
+		throw new Error(`Unable to load files of type "${type}"`)
+	}
+}
+
+function loadJSON(url) {
+	return loadXhr(url, (data) => {
+		if (typeof data === 'string') {
+			return JSON.parse(data)
+		}
+
+		return data;
+	});
+}
+
+function loadXhr(url, dataParse = (data) => {return data}) {
+	let request = new XMLHttpRequest();
```

Comment:
Waarom niet fetch ipv xmlhttprequest? :)

Reply:
Mjoah, niet superveel support, maar ik weet nog niet of ik daar perse rekening mee wil houden. `fetch` doet wel gelijk promises, wat ik hier nu soort van probeer te simuleren, dus dat zou wel fijn zijn.

Reply:
Ik ben wel zo'n hater die gewoon niet alles support of gewoon 2mb aan polyfills naar oude browsers stuurt 😂

Reply:
Haha ja soms heb ik ook wel schijt, en ik zie nu dat `fetch` vooral IE mist, dus wat dat betreft kan ik die er wel in yolo'en.